vendor/github.com/samuel/go-zookeeper/zk/dnshostprovider.go

```diff
@@ -0,0 +1,88 @@
+package zk
+
+import (
+	"fmt"
+	"net"
+	"sync"
+)
+
+// DNSHostProvider is the default HostProvider. It currently matches
+// the Java StaticHostProvider, resolving hosts from DNS once during
+// the call to Init.  It could be easily extended to re-query DNS
+// periodically or if there is trouble connecting.
+type DNSHostProvider struct {
```

Comment:
DNS clients tend to be dumb, we have no idea stale data is being served, there is no alerting / monitoring of any of this. It's a silent failure that shows up in bad behavior. Failing hard at least makes a nice, recognizable impact of where to look, and shows up in the monitoring we have today.